project01/dominik/game.cc

```diff
@@ -0,0 +1,65 @@
+#include "game.h"
+
+static const int REWARD_FOR_BAIT = 10;
+
+Game::Game(const cinder::ivec2& screenSize, const int snakeBodyPartSize)
+    : mFieldSize(screenSize),
+      mSnake(mFieldSize, snakeBodyPartSize),
+      mBait(mFieldSize),
+      mState(GameState::running),
+      mScore(0)
+{}
+
+void Game::Update()
+{
+    mSnake.Move();
+
+    if (mSnake.HeadClashedWithBody() || mSnake.HeadOutsideField())
+    {
+        std::cout << "Head clashed with body or outside field!" << std::endl;
+        mState = GameState::lost;
+        return;
+    }
+    if (mSnake.MinDistanceHeadToVectorNowAndBefore(mBait.GetPosition()) < static_cast<float>(mBait.GetRadius() + mSnake.GetSize()))
```

Comment:
To detect that snakes ate the fruit, why not using a simple logic, identical as you implemented for detecting if snake collided with its own body? Whats teh difference?